Avoid strncat string overflow warning

hornungt opened the pull request:
Previously got an error about string overflow due to strncat usage

error: ‘__builtin___strncat_chk’ specified bound depends on the length of the source argument [-Werror=stringop-overflow=]

Replace strncat with snprintf to remove the error

hornungt committed:
Avoid strncat string overflow warning

Previously got an error about string overflow due to strncat usage

error: ‘__builtin___strncat_chk’ specified bound depends on the
length of the source argument [-Werror=stringop-overflow=]

Replace strncat with snprintf to remove the error

ion201 commented on src/fscrypt_utils.c:
```diff
@@ -351,11 +351,18 @@ enum fscrypt_utils_status_t lock_unlock_data_file(const char *datapath, int lock
-    strncat(lock_path, LOCK_FILE_SUFFIX, lock_path_length - strlen(lock_path) - 1);
+    if (NULL == lock_path)
+    {
+        fscrypt_utils_log(LOG_INFO, "Failed to allocate lock path string");
```
This should be `LOG_ERR`

hornungt committed:
Upgrade fail message from INFO to ERR

Upgrade the fail message for failure to allocate the lock_path string in
lock_unlock_data_file
